src/utilities/modal.ts

```diff
@@ -143,7 +159,188 @@ export async function showConfirmModal(
 }
 
 /**
- * Shows an ManagePlanConstraintsModal component with the supplied arguments.
+ * Shows a CreateGroupsOrTypesModal component with the supplied arguments.
+ */
+export async function showCreateGroupsOrTypes(user: User | null): Promise<ModalElementValue> {
+  return new Promise(resolve => {
+    if (browser) {
+      const target: ModalElement | null = document.querySelector('#svelte-modal');
+
+      if (target) {
+        const manageGroupsAndTypesModal = new CreateGroupsOrTypesModal({
+          props: { user },
+          target,
+        });
+        target.resolve = resolve;
+
+        manageGroupsAndTypesModal.$on('close', () => {
+          target.replaceChildren();
+          target.resolve = null;
+          target.removeAttribute('data-dismissible');
+          manageGroupsAndTypesModal.$destroy();
+        });
+        manageGroupsAndTypesModal.$on('add', (e: CustomEvent<{ derivationGroupName: string }[]>) => {
+          target.replaceChildren();
+          target.resolve = null;
+          resolve({ confirm: true, value: e.detail });
+          manageGroupsAndTypesModal.$destroy();
+        });
+      }
+    } else {
+      resolve({ confirm: false });
+    }
+  });
+}
+
+/**
+ * Shows a DeleteExternalSourceModal component with the supplied arguments.
+ */
+export async function showDeleteExternalSourceModal(
+  linked: { pkey: ExternalSourcePkey; plan_ids: number[] }[],
+  sources: ExternalSourceSlim[],
+  unassociatedSources: ExternalSourceSlim[],
+): Promise<ModalElementValue> {
+  return new Promise(resolve => {
+    if (browser) {
+      const target: ModalElement | null = document.querySelector('#svelte-modal');
+
+      if (target) {
```

Comment:
Might just be a matter of preventing default on that button? But fine to leave as is if not.

Reply:
(Per slack conversation - no action required.)